### ISSUE TITLE
Test how to manage translations for multiple Slicer versions

### ISSUE THREAD

**Issue by lassoan** [Slicer/SlicerLanguagePacks#6]
Main options:
- A. Use git branches
- B. Use subfolders

Create a test project on weblate to compare the options.

**Comment by lassoan** [comment]
@comradekingu Can you advise us how to manage translations of multiple software versions in weblate?

If we run the `lupdate` tool then it replaces old strings by new ones but keeps the previous source text in an `oldsource` tag. 
See for example source text changed from `Visible` to `Visibility`:

```
    <message>
        <location filename="Modules/Loadable/Colors/Widgets/Resources/UI/qMRMLColorLegendDisplayNodeWidget.ui" line="36"/>
        <source>Visibility:</source>
        <oldsource>Visible:</oldsource>
        <translation type="unfinished"></translation>
    </message>
```

Is weblate going to use this `oldsource` string? Is weblate going to remember the old source string and its translations somewhere?

We plan to create a new release of our application and so we'll have translations for our latest stable (Slicer-5.0) and latest preview (Slicer-5.1) versions. How would you recommend to store these two translation versions in the repository? It would be nice if translators would not need to submit translations to both versions independently, but translation of the same source string would be synchronized between versions.

**Comment by comradekingu** [comment]
Oldsource might end up in the history for a string?

Having two versions in separate components can share translations between them.
Anything other than a 100% match will be fuzzy (selectable).
Spreading of tranlsations is on in https://hosted.weblate.org/settings/3d-slicer/3d-slicer/#translation

https://docs.weblate.org/en/latest/workflows.html#quality-gateway-for-the-source-strings is somewhat related.

**Comment by lassoan** [comment]
Where should we store the new version's source strings?

Right now we don't use git branches and the source strings are stored in the file `master/Slicer.ts` (there will be a few dozens more .ts files).

Should we store the source strings in a new folder, such as `5.0/Slicer.ts`?
Or should we start using git branches for versioning, such as use the same `source/Slicer.ts` file path in github branches `master` and `5.0`?

**Comment by comradekingu** [comment]
I think Git branches is better for reasons other than Weblate caring about it, as it makes it easy to push versions where history is kept uninterrupted, and people can get an overview of what is going on without exploring file structures or reading.

In any event,
https://hosted.weblate.org/settings/3d-slicer/3d-slicer/#files for filemask/file
https://hosted.weblate.org/settings/3d-slicer/3d-slicer/#vcs for branch

**Comment by lassoan** [comment]
Thank you for the advice. At https://hosted.weblate.org/settings/3d-slicer/3d-slicer/#vcs I see that I can only specify a single branch. Does this mean that I would need to specify a new component for each version (master, 5.0, 5.1, 5.2, ...)?

I also have questions about https://hosted.weblate.org/settings/3d-slicer/3d-slicer/#files that you have kindly set up for us. `Template for new translations` is set to `translated/master/Slicer_en-US.ts`, which is in the same folder as the translated files (https://github.com/Slicer/SlicerLanguageTranslations/blob/main/translated/master/Slicer_en-US.ts). But we also have a similar en-US source translation file in the `master` folder (https://github.com/Slicer/SlicerLanguageTranslations/blob/main/master/Slicer.ts). **Could we remove one of these files? Which one should we keep?** For me it seems a bit more clean for me to separate the source file from the translated files, but if it is more common to keep all the files in the same folder then that should be fine, too.

**Comment by Pullpe** [comment]
Hello,
I have trouble with the "file mask" field when adding a translation component to the test project on Weblate. 
I used the path to the translation file on the git repository but perhaps I put the " * " character in the wrong place or maybe this isn't the path I should be using.
I get "The file mask did not match any files." as an error.
I also tried using the same file mask as the 3D Slicer project ("translated/master/Slicer_*.ts") but I get the same error.
Could you suggest solutions?

**Comment by lassoan** [comment]
Can you add me as an administrator to your test project so that I can see your project settings?

**Comment by Pullpe** [comment]
Okay, done

> Can you add me as an administrator to your test project so that I can see your project settings?

**Comment by lassoan** [comment]
We'll manage translation of different Slicer versions in branches.

I've created a branch for Slicer-5.0 (https://github.com/Slicer/SlicerLanguageTranslations/tree/5.0) once Slicer-5.2 is released, I'll create one for Slice-5.2.

I've updated all the source strings from Slicer source code (using Utilities\Scripts\update_translations.py in Slicer repository). Weblate nicely took the changes from the github repository and updated translations on the Weblate website. It reduced the previously 100% translated languages to 88% (due to new strings were added or strings were changed). I'll keep updating source strings from Slicer source code more frequently.
